mlflow/utils/rest_utils.py

```diff
@@ -30,11 +32,27 @@
 )
 from mlflow.utils.string_utils import strip_suffix
 
+
 RESOURCE_NON_EXISTENT = "RESOURCE_DOES_NOT_EXIST"
 _REST_API_PATH_PREFIX = "/api/2.0"
 _TRACE_REST_API_PATH_PREFIX = f"{_REST_API_PATH_PREFIX}/mlflow/traces"
 
 
+@dataclass
+class _DatabricksSdkAPIErrorResponse:
+    status_code: int
+    reason: str
+    text: str
+
+    def raise_for_status(self):
+        if self.status_code == 200:
+            return
+
+        http_error_msg = f"{self.status_code} Error: {self.reason}"
+
+        raise HTTPError(http_error_msg)
```

Comment:
We should not do this. Instead, we should inherit requests' response object.

Reply:
oh I got it, let me try to inherit requests' Response object

Reply:
Rewrite class as:

```
class _DatabricksSdkAPIErrorResponse(requests.Response):
    def __init__(self, status_code, reason, content):
        super().__init__()
        self.status_code = status_code
        self.reason = reason
        self.encoding = "UTF-8"
        self._content = content.encode(self.encoding)
```